simson/steel/inflow_driven_steel_historic.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+from numpy.linalg import inv
+
+from sodym import MFASystem
+
+from simson.common.inflow_driven_mfa import InflowDrivenHistoricMFA
+from simson.common.common_cfg import CommonCfg
+
+class InflowDrivenHistoricSteelMFASystem(InflowDrivenHistoricMFA):
+
+    """def __init__(self, cfg:CommonCfg, **kwargs):
+        super(MFASystem, self).__init__(**kwargs)
+        self.cfg = cfg
+        print('Test')"""
+
+
+    def compute(self):
+        """
+        Perform all computations for the MFA system.
+        """
+        self.compute_historic_flows()
+        self.compute_historic_in_use_stock()
+        self.check_mass_balance()
+
+
+    def compute_historic_flows(self):
+        prm = self.parameters
+        flw = self.flows
+        stk = self.stocks
+        scp = self.scalar_parameters
+
+        aux = {
+            'net_intermediate_trade': self.get_new_array(dim_letters=('h','r','i')),
+            'fabrication_by_sector': self.get_new_array(dim_letters=('h','r','g')),
+            'fabrication_loss': self.get_new_array(dim_letters=('h','r','g')),
+            'fabrication_error': self.get_new_array(dim_letters=('h','r'))
+        }
+
+        flw['sysenv => forming'][...] = prm['production_by_intermediate']
+        flw['forming => ip_market'][...] = prm['production_by_intermediate'] * prm['forming_yield']
+        flw['forming => sysenv'][...] = flw['sysenv => forming'] - flw['forming => ip_market']
+
+        # Todo: add trade
+
+        aux['net_intermediate_trade'][...] = flw['ip_trade => ip_market'] - flw['ip_market => ip_trade']
+        flw['ip_market => fabrication'][...] = flw['forming => ip_market'] + aux['net_intermediate_trade']
+
+        aux['fabrication_by_sector'][...] = self._calc_sector_flows(
+            flw['ip_market => fabrication'],
+            prm['good_to_intermediate_distribution'])
+
+        aux['fabrication_error'] = flw['ip_market => fabrication'] - aux['fabrication_by_sector']
+
+        flw['fabrication => use'][...] = aux['fabrication_by_sector'] * prm['fabrication_yield']
+        aux['fabrication_loss'][...] = aux['fabrication_by_sector'] - flw['fabrication => use']
+        flw['fabrication => sysenv'][...] = aux['fabrication_error'] + aux['fabrication_loss']
+        # Todo: add intermediate trade
+
+        return
+
+    def _calc_sector_flows(self, intermediate_flow, gi_distribution):
+        """
+        Estimate the fabrication by in-use-good according to the inflow of intermediate products
+        and the good to intermediate product distribution.
+
+        See https://en.wikipedia.org/wiki/Overdetermined_system#Approximate_solutions
+        """
+
+
+        gi_values = gi_distribution.values.transpose()
+        at_a = np.matmul(gi_values.transpose(), gi_values)
```

Comment:
sure, definitely. I wasn't really sure how though - gi_values I can definitely rename, at_a here means matrix a transposed times matrix a, basically I am solving for a parameter here according to a standard optimization approximation Jakob proposed (to minimize Ax-b). I guess the options are to either to 
1. squish all the lines into one to avoid using that variable at all
2. Come up with some sort of variable name that describes this
3. Keep it as it is but make a better explanatory comment before
What do you think would be best? I'm tending towards option 3, but am happy to go with something else (I know it probably doesn't matter too much but wanted to ask cause I don't have a good feel for how to write very self-explanatory code yet).

Reply:
Okay, yes it's difficult when there isn't really a physical meaning, and squashing all of that into line 74 is probably too much. I also think a comment is the best option.